docs/workflows/best-practices-architecture.md

```diff
@@ -0,0 +1,76 @@
+
+When it comes to defining components, consistency can be a double-edged sword. Defining functional components has benefits, but it’s almost impossible to define such a structure for your entire system. You’re almost inevitably going to end up with a `common` or `shared` or `utility` component that contains cross-cutting concerns used by all functional components. The outcome is a mixed structure, where most of your components are defined based on functional boundaries, but you also have a small number of technical components. This is fine, having 100% consistency across all components is not a goal in itself.
+
+## When is a component too big?
+
+As we’ve seen in the previous section, the most common reason for needing to revisit your modularization is one of your components becoming too big. 
+
+For example, consider the following component structure:
+
+<img src="../images/methodology/components-example-4.png" width="250" />
+
+Is this a logical top-level component structure? Definitely. Is this a suitable modularization? That depends on the size of the components. If both components are 5,000 lines of code, this is perfectly suitable. But if both components contain 20 million lines of code, this component structure is no longer suitable. 
+
+If the original modularization is never revisited, you might end up in a situation where components become excessively large. In such a situation, introducing an additional layer of sub-components helps to guide the components’ evolution. It preserves the original structure, without making components overwhelming due to lack of internal structure.
+
+<img src="../images/methodology/components-example-5.png" width="600" />
+
+Introducing sub-components also helps to manage dependencies between the sub-components. Again, this is not really an issue if the component is small, but it can become problematic if the amount of code in the component continues to grow.
+
+## How should you manage evolving components?
+
+The initial component structure is usually designed at the beginning of the project, often before the first line of code has been written. A common pitfall is to basically retain this initial structure forever, and never revisit it. Over the years, systems tend to involve, and the original component structure is not necessarily designed to handle those changes. 
+
+A suitable component structure tends to look like a tree. Defining 100 top-level components is not particularly helpful for locating functionality, so even very large systems should strive towards a manageable number of top-level components. However, larger systems tend to have more levels of sub-components. Every component has its own internal structure, and as components continue to grow, they tend to spawn additional levels of sub-components to manage their internal structure. 
```

Comment:
perhaps too detailed, but since we mentioned components with crosscutting concerns; those break the tree structure, turning it into a lattice. perhaps we can add "... like a tree (with some shared subcomponents)." as a simple way to add this universal exception to the rule (..)